src/test/java/org/opensearch/knn/index/OpenSearchIT.java

```diff
@@ -900,6 +901,136 @@ public void testKNNIndex_whenBuildVectorGraphThresholdIsProvidedEndToEnd_thenBui
         deleteKNNIndex(indexName);
     }
 
+    public void testKNNIndexSearchFieldsParameter() throws Exception {
```

Comment:
Also lets add an IT test where some the docs don't have vector field and some have vector field in it.